system-metrics/handler.go

```diff
@@ -0,0 +1,125 @@
+package function
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+
+	"github.com/openfaas/faas/gateway/metrics"
+)
+
+type Metrics struct {
+	Success int `json:"success"`
+	Failure int `json:"failure"`
+}
+
+// Handle a serverless request
+func Handle(req []byte) string {
+	fnName, err := parseFunctionName()
+
+	if err != nil {
+		log.Fatalf("couldn't parse function name from query: %t", err)
+	}
+
+	host := os.Getenv("prometheus_host")
+	envPort := os.Getenv("prometheus_port")
+	port, err := strconv.Atoi(envPort)
+	if err != nil {
+		log.Fatalf("Could not convert env-var prometheus_port to int. Env-var value: %s. Error: %t", envPort, err)
+	}
+
+	metricsQuery := metrics.NewPrometheusQuery(host, port, &http.Client{})
+	metricsWindow := os.Getenv("metrics_window")
+
+	if metricsWindow == "" {
+		metricsWindow = "60m"
+	}
+
+	fnMetrics, err := getMetrics(fnName, metricsQuery, metricsWindow)
+	if err != nil {
+		log.Fatalf("Couldn't get metrics from Prometheus for function %s, %t", fnName, err)
+	}
+
+	res, err := json.Marshal(fnMetrics)
+	if err != nil {
+		log.Fatalf("Couldn't marshal json %t", err)
+	}
+	return string(res)
+}
+
+func parseFunctionName() (functionName string, error error) {
+	if query, exists := os.LookupEnv("Http_Query"); exists {
+		vals, _ := url.ParseQuery(query)
+
+		functionNameQuery := vals.Get("function")
+
+		if len(functionNameQuery) > 0 {
+			return functionNameQuery, nil
+		} else {
+			return "", fmt.Errorf("there is no `function` inside env var Http_Query")
+		}
+	}
+
+	return "", fmt.Errorf("env var Http_Query because it doesn't exist")
+}
+
+func getMetrics(fnName string, metricsQuery metrics.PrometheusQueryFetcher, metricsWindow string) (*Metrics, error) {
+	queryValue := fmt.Sprintf(
```

Comment:
Probably break out a small function for this on its own so that you can unit test it as part of this PR.